scripts/ci/rust_checks.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python3
+
+"""Run various rust checks for CI."""
+
+from __future__ import annotations
+
+import os
+import re
+import subprocess
+import sys
+import time
+
+
+class Timing:
+    def __init__(self, command: str, start_time: float) -> None:
+        self.command = command
+        self.duration = time.time() - start_time
+
+
+def run_cargo(args: str) -> Timing:
+    command = f"cargo {args}"
```

Comment:
I suggest passing in `--quiet` to all the runs to reduce the amount of log spam, especially when running locally

Reply:
hard to see progress then though ⚖️ 
but don't mind too much. makes the timing overview less critical as well